fix: recently added items bug fixed where you could not see newly add…

Edga380 opened the pull request:
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review on my code
- [x] I have made corresponding changes to the documentation (if any were needed)
- [x] My changes generate no new warnings or errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and previously existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Description

** Closes #355 **  
Fixes a bug where you could not see newly added items on the home page.

### Files changed

- ### **app/page.tsx**
  - Replaced recently added items code with a newly created component that is imported. `<DisplayRecentlyAddedItems />`

- ### **supabase/models/getRecentItems.ts**
  - Supabase now uses a custom function instead of built in methods reason for this is that these methods just didn't work as intended for example if I removed `limit(4)` still some items were missing from the array.
  - Added try/catch so we could catch errors.
  - Added types for the returned result from database.
  - Function is exported as `default` now.

### New files

- ### **supabase/migrations/fetch-recently-added-items.sql**
  - Like I mentioned before built in supabase methods didn't work as expected so I created new supabase function and migration file.
  - This function returns `items.id`, `items.created_at`, `items.item_name`, `items."imageSrc"` where `items.given_away_to` is `null` and `items.is_reserved` is `false` I thought that users should not see if the item is already given away or it is reserved by someone. You wouldn't want to click on an item and see that you can't reserve it.
  - Returns `4` items.

- ### **components/DisplayRecentlyAddedItems.tsx**
  - I created this component mainly so I could utilize `useEffect` hook to fetch latest data from database.
  - User sees `Loding...` text while we wait for the data if we receive data we display items if array length is longer than 0 otherwise we display message `No added items`.

- ### **components/LoadingModal.tsx**
  - Displays 'Loading...' message while data is being fetched.

### UI changes

None

### Changes to Documentation

None

# Tests

- ### **cypress/e2e/anonymous.cy.js**
  - I decided to add this test to this specific file because I didn't want to create new one for a small test like this and I felt like it belongs there.
  - `Can see recently added items` checks if we received data by checking if div element with `className=".lastItems"` do not contain `Loading...` text (There is delay to make sure that we wait for the data) if we pass this point then we check do we have item element array by checking length if length is not more than 0 then we check if div element `className=".lastItems"` contain text `No added items`.

ShaughnAnderson94 reviewed (changes requested):
We try and keep comments out of the code base can you go through and remove the comments form this code. Mostly its looking good though

Edga380 commented:
@ShaughnAnderson94 I added LoadingModal.tsx file and implemented it to work with DisplayRecentlyAddedItems.tsx also I have tested it with DonatedItemsList.tsx and it works without an issue.

ShaughnAnderson94 reviewed (approved):
Ok this looks better!
